### ISSUE TITLE
Can't update to 5.15.15 but can work with linux-image-edge-arm64 (5.15.16) (on SD-Card)

### ISSUE THREAD

**Issue by MX10-AC2N** [ophub/amlogic-s9xxx-armbian#84]
Hi @ophub 
I'm opening a new issue because I'm not sure you've seen my last posts in https://github.com/ophub/amlogic-s9xxx-armbian/issues/72#issuecomment-1016577181
So long story short, I managed to install and boot linux-image-edge-arm64 (kernel 5.15.16) from my sd card and so I wonder why "armbian-update 5.15.15" is giving me a problem during boot, there is a problem in the config since linux-image-edge-arm64 works well..
Hope this can help, I can still test if needed..
Thanks again..

**Comment by ophub** [comment]
I downloaded the firmware generated today, tested it on TF and it works fine. This is the HG680P box that has just been flashed with new firmware.

![Snip20220122_4](https://user-images.githubusercontent.com/68696949/150632676-8cca7a78-78db-401d-8c20-00a89c43048c.png)

**Comment by MX10-AC2N** [comment]
I could boot 5.15.15 from usb without problem but from sd card it doesn't work.. but with linux-image-edge-arm64 it works..
Since I have a resistor in place between Rx and GND I can now use u-boot-gtking-pro.bin as u-boot.ext
Do you think I can try an "armbian-update 5.15.16" ?

**Comment by ophub** [comment]
I tested it on the s905d and s905x3 boxes again, and there is no problem. This is the system `installed in the emmc`.

![Snip20220122_6](https://user-images.githubusercontent.com/68696949/150633389-857e16b6-6edf-4580-bf28-9fc8c105ba25.png)
![Snip20220122_7](https://user-images.githubusercontent.com/68696949/150633390-3c201f8b-26e3-4483-8b4b-d2795fe6d6b1.png)

**Comment by ophub** [comment]
You can upgrade the local kernel update script, the latest update script can directly input armbian-update to find the latest kernel update.

`wget -O /usr/sbin/armbian-update git.io/armbian-update`

**Comment by MX10-AC2N** [comment]
I'm not sure I understand correctly, in my case adding the resistor finally allows you to be able to use u-boot-gtking-pro.bin as u-boot.ext without there being a freeze at the start of the boot, without this resistance impossible to use u-boot-gtking-pro.bin and unfortunately the u-boot-gtking-pro-rev_a.bin only works if I boot from a USB medium, for my part j I need to install the system on SD Card in order to keep my USB inputs (used by 2 HDD, 1 SSD and the remote control receiver)
So for the moment this addition of resistance is very useful to me..
I still specify that I kept an Android system on my Emmc (this probably explains why the addition of the resistor is useful to me..) would it be possible to create a command allowing the reboot to android as we can have under Coreelec.
I will try an update to 5.15.16 to see if it works.
Thank you again for your investment in this project. You're great..

**Comment by ophub** [comment]
When the box was just bought back last week, I made a ddbr backup, and then wrote emmc. I used u-boot-gtking-pro-rev_a.bin, and did not write the mainline u-boot, after writing emmc It works normally. I used it for a few days, and then ddbr went back to the Android system. Now I use it with the usb plugged into the usb port of the OTG.

I also tested u-boot-gtkingpro.bin, and also tested writing the mainline u-boot to emmc. I tried various combinations of several u-boots for s922x in the warehouse and flashed them into bricks. . But in the end, there is only one solution that my box can write to emmc, that is, do not use the mainline u-boot, use `u-boot-gtking-pro-rev_a.bin` as the boot, and use the `meson-g12b-gtking-pro.dtb` file .

**Comment by MX10-AC2N** [comment]
Well we can say that Beelink has added a large heat sink, on my GT-King Pro card there is none..
We don't have the same SBC at all.
Either I did armbian-update 5.10.93 like that I went back to kernel 5.10.93, the reboot was successful and I therefore launched armbian-update 5.15.16 and again I have this problem `I/O error, dev mmcblk1` => https://imgur.com/a/F6eSJUq
There must be some setting wrong with flippy kernel 5.15 since linux-image-edge-arm64 kernel 5.15.16 boots fine from the same SD card without me changing the u-boot.ext (u- boot-gtking-pro.bin)

**Comment by ophub** [comment]
Then you have been using the 5.10 kernel. After 93 revisions, it is very stable.
5.15 kernel and so on. In the use of openwrt, 5.10 and 5.4 are also the most stable.
I thought about it for a long time, and only changed the document to 5.15 last night. I just decided to use 5.15 as the default kernel, which seems to be hasty.

**Comment by MX10-AC2N** [comment]
I'm really not an expert with this but if my feedback can help you I would be delighted.
It's a fair return for everything you do.
I take this opportunity to add a little thing, would it be possible to keep a 5.10.y backup during the update to 5.15.y so as to be able to return to 5.10.y if there was a problem with 5.15.y because for the moment when I do the update the files relating to 5.10.y are deleted from /BOOT and therefore I have to re-flash my sd-card from scratch..
thank you very much again

**Comment by ophub** [comment]
I think if updating the kernel fails, what is the simple and easy to use fallback recovery method

**Comment by MX10-AC2N** [comment]
I just managed to boot kernel 5.15.16 only once from my SD card, I managed to boot just using gtking-pro.dtb from linux-image-edge-arm64 successful boot but loss of hdmi signal just after starting the kernel, I was able to connect in SSH unfortunately the reboot did not work. And despite several attempts I can no longer have SSH access.
It seems that the problem is related to the dtb file..
Here is a copy of gtking-pro.dtb taken from linux-image-edge-arm64 maybe that will help find the problem
[meson-g12b-gtking-pro-arm64.zip](https://github.com/ophub/amlogic-s9xxx-armbian/files/7918285/meson-g12b-gtking-pro-arm64.zip)
.

**Comment by ophub** [comment]
There are small version differences in the box, you still use the 5.10 kernel, it is stable

**Comment by JFLim1** [comment]
> The two points shown in the figure below are the `shorting points` of the `recovery android tv system`.
> 
> ![IMG_1322](https://user-images.githubusercontent.com/68696949/150634573-c091cb81-f702-47c2-8d86-b22a0610765e.jpg)

A large heatsink in addition to full aluminium casing. Good.

**Comment by MX10-AC2N** [comment]
`Is your model come with Wifi 6 and Wifi 6 driver available for mainline kernel?`
Hi @ophub
No, unfortunately my GTKing-pro is not Wifi6 it is one of the first versions of the SBC with an RS232 port which one wonders what it could be used for...
( For Wifi6 drivers it seems to me that @pulpoff must know, he uploaded https://vp5.net/king/BCM4362A2.hcd ) 
Another thing, I went back to kernel 5.10.93 but when I do the reboot command the box restarts well but loses the hdmi signal at the start of the kernel while with the poweroff command and a restart by pressing the button on the box works well without losing the hdmi signal.. are there a few things to modify in the reboot command?
have a nice day..

**Comment by MX10-AC2N** [comment]
`Is your model come with Wifi 6 and Wifi 6 driver available for mainline kernel?`
Hi @ophub
No, unfortunately my GTKing-pro is not Wifi6 it is one of the first versions of the SBC with an RS232 port which one wonders what it could be used for...
For Wifi6 drivers it seems to me that @pulpoff must know, he uploaded https://vp5.net/king/BCM4362A2.hcd
https://4pda.to/forum/index.php?showtopic=948258&view=findpost&p=111152414

Another thing, I went back to kernel 5.10.93 but when I do the reboot command the box restarts well but loses the hdmi signal at the start of the kernel while with the poweroff command and a restart by pressing the button on the box works well without losing the hdmi signal.. are there a few things to modify in the reboot command?
have a nice day..
